trust-sdk-client/src/main/java/trust/Trust.kt

```diff
@@ -21,19 +21,46 @@ object Trust {
             activity.startActivityForResult(intent, request.getRequestCode())
             Call(request)
         } else {
-            try {
-                activity.startActivity(Intent(
-                        Intent.ACTION_VIEW,
-                        Uri.parse("market://details?id=$packageName")))
-            } catch (anfe: ActivityNotFoundException) {
-                activity.startActivity(Intent(
-                        Intent.ACTION_VIEW,
-                        Uri.parse("https://play.google.com/store/apps/details?id=$packageName")))
-            }
+            openMarket(activity)
             null
         }
     }
 
+    fun execute(activity: Activity, operation: Operation) {
+        val intent = Intent(Intent.ACTION_VIEW, operation.buildUri())
+        if (canStartActivity(activity, intent)) {
+            activity.startActivity(intent)
+        } else {
+            openMarket(activity)
+        }
+    }
+
+    fun handleTransactionResult(intent: Intent?): TransactionResult? = if (intent?.data?.host == Host.TX_CALLBACK.key) {
+        val signature = intent.data?.getQueryParameter(ExtraKey.TRANSACTION_SIGN.key)
+        val hash = intent.data?.getQueryParameter(ExtraKey.TRANSACTION_HASH.key)
+        val error = intent.data?.getQueryParameter(ExtraKey.CANCEL.key)
+        when {
+            signature != null -> TransactionResult(signature = signature)
+            hash != null -> TransactionResult(hash = hash)
+            error != null -> TransactionResult(isCancelled = true)
+            else -> null
+        }
+    } else {
+        null
+    }
+
+    private fun openMarket(activity: Activity) {
+        try {
+            activity.startActivity(Intent(
+                    Intent.ACTION_VIEW,
+                    Uri.parse("market://details?id=${packageName}")))
+        } catch (ignored: ActivityNotFoundException) {
+            activity.startActivity(Intent(
```

Comment:
Could we open trustwallet.com if market doesn't make available?

Reply:
we use this, because package name is customizable